lib/perl/Genome/Config/Profile.pm

```diff
@@ -188,14 +189,35 @@ sub _value_for_instrument_data_property {
     if (ref $instrument_data_property eq 'ARRAY') {
         return [
             grep { defined($_) }
-            map { $instrument_data->$_ }
+            map { $self->_apply_method_chain_for_instrument_data_property($instrument_data,$_) }
             @$instrument_data_property
         ];
     } else {
-        return $instrument_data->$instrument_data_property;
+        return $self->_apply_method_chain_for_instrument_data_property($instrument_data,$instrument_data_property);
     }
 }
 
+sub _apply_method_chain_for_instrument_data_property {
+    my ($self, $instrument_data, $instrument_data_property) = @_;
+
+    unless ($instrument_data_property) {
+        die 'bad call to _apply_method_chain_for_instrument_data_property: missing instrument_data_property';
+    }
+    my @chain = split('->', $instrument_data_property);
+
+    my $current_result = $instrument_data;
+
+    try {
+        for my $next_property (@chain) {
+            $current_result = $current_result->$next_property;
```

Comment:
I didn't know about the delete thing!  Is that something we should guard against here?  

    die('"delete" is not a valid property') if $next_property eq "delete";

Reply:
If it's likely someone will accidentally specify `delete` as the property they want in a configuration, then absolutely we could add that.

If it's intended to stop malicious behavior, this would be a lot more roundabout of a strategy than any of the other myriad available ways of deleting bunches of data.